erasure-readfile.go

```diff
@@ -159,7 +159,7 @@ func parallelRead(volume, path string, readDisks []StorageAPI, orderedDisks []St
 // are decoded into a data block. Data block is trimmed for given offset and length,
 // then written to given writer. This function also supports bit-rot detection by
 // verifying checksum of individual block's checksum.
-func erasureReadFile(writer io.Writer, disks []StorageAPI, volume string, path string, offset int64, length int64, totalLength int64, blockSize int64, dataBlocks int, parityBlocks int, checkSums []string, pool *bpool.BytePool) (int64, error) {
+func erasureReadFile(writer io.Writer, disks []StorageAPI, volume string, path string, offset int64, length int64, totalLength int64, blockSize int64, dataBlocks int, parityBlocks int, checkSums []checkSumInfo, pool *bpool.BytePool) (int64, error) {
 	// Offset and length cannot be negative.
```

Comment:
For erasureReadFile also, it's better to pass `algo string` and `checkSums []string` separately than passing `[]checkSumInfo` as it expresses the function dependency explicitly than to pass checkSumInfo which has the partName also which is not needed by erasureReadFile.

passing []checkSumInfo makes the code reader of erasureReadFile believe that each disk will have different algo

checkSumInfo belongs to a higher layer as it contains partName which is not used by erasureCreateFile and erasureReadFile